kingpin/actors/rightscale/alerts.py

```diff
@@ -0,0 +1,364 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2014 Nextdoor.com, Inc
+
+"""
+:mod:`kingpin.actors.rightscale.alerts`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+"""
+
+import logging
+
+from tornado import gen
+import requests
+
+from kingpin.actors import exceptions
+from kingpin.actors.rightscale import base
+from kingpin.constants import REQUIRED
+
+log = logging.getLogger(__name__)
+
+__author__ = 'Matt Wise <matt@nextdoor.com>'
+
+
+class InvalidInputs(exceptions.InvalidOptions):
+
+    """Raised when supplied inputs are invalid for a ServerArray."""
+
+
+class AlertSpecNotFound(exceptions.RecoverableActorFailure):
+
+    """Raised when an Alert Spec could not be found"""
+
+
+class CreateFailed(exceptions.RecoverableActorFailure):
+
+    """Raised when an Alert Spec could not be created"""
+
+
+class AlertsBaseActor(base.RightScaleBaseActor):
+
+    """Abstract Alerts Actor that provides some utility methods."""
+
+    @gen.coroutine
+    def _find_alert_spec(self, name, subject_href):
+        """Search for an AlertSpec by-name and return the resource.
+
+        Note: A non-exact resource match is used below so that we return all of
+        the Alert Specs that are matched by name. This method returns the
+        resources in a list.
+
+        Args:
+            name: RightScale AlertSpec Name
+            subject_href: The HREF of the subject this AlertSpec is assigned
+              to.
+
+        Return:
+            [<rightcale.Resource objects>]
+        """
+        log.debug('Searching for AlertSpec matching: %s' % name)
+        found_spec = yield self._client.find_by_name_and_keys(
+            self._client._client.alert_specs, name, exact=False,
+            subject_href=subject_href)
+
+        if not found_spec:
+            log.debug('AlertSpec matching "%s" could not be found.' % name)
+            return
+
+        log.debug('Got AlertSpec: %s' % found_spec)
+        raise gen.Return(found_spec)
+
+
+class Create(AlertsBaseActor):
+
+    """Create a RightScale Alert Spec
```

Comment:
Add a link to RightScale API describing the expected parameters. Seems that our option names are identical to the ones required by the API